### PR TITLE
chore(pods): migrated json export pod to v2

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
# Pull Request Checklist

If this is your first time submitting a pull request to Dendron, copy and paste the full [Dendron Review Checklist](https://docs.dendron.so/notes/1EoNIXzgmhgagqcAo9nDn.html) into this request and check off each item as necessary. 

This template contains the short checklist which is used by the Dendron core team. 

## Testing
- [x] [Write Tests](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#writing-tests) 
- [x] [Confirm existing tests pass](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#executing-tests)
- [x] [Confirm manual testing](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#manual-testing) 
- [ ] If your tests changes an existing snapshot, make sure that snapshots are [updated](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#updating-test-snapshots)
- [ ] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), make sure that it is included in the [test workspace](https://docs.dendron.so/notes/dtMsF12SF2SUhLN10sYe2.html)
### Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
### Analytics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://docs.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### CODE REVIEW

File: packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+} from "@dendronhq/common-all";
+import { JSONSchemaType } from "ajv";
+import _ from "lodash";
+import path from "path";
+import fs from "fs-extra";
+import { ConfigFileUtils, ExportPodV2 } from "../../..";
+import { JSONV2PodConfig, RunnableJSONV2PodConfig } from "../..";
+
+/**
+ * Markdown Export Pod (V2 - for compatibility with Pod V2 workflow).
```
Comment (jonathanyeung) [inline packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts:15]: nit: change 'markdown'

File: packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ */
+
+export type JSONExportReturnType = RespV2<{
+  exportedNotes?: string | NoteProps[];
```
Comment (jonathanyeung) [inline packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts:19]: we can remove `string` now right?

File: packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    });
+  }
+
+  static config(): JSONSchemaType<JSONV2PodConfig> {
+    return ConfigFileUtils.createExportConfig({
+      required: ["destination"],
+      properties: {
+        destination: {
+          description:
+            "export destination. Specify either a file path or 'clipboard' to export to your clipboard",
```
Comment (jonathanyeung) [inline packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts:67]: What happens if the user has a config that specifies exportScope to be multi-note (i.e. hierarchy), and destination == 'clipboard'?  Will we error out w/ a useful error message? 

Can we also add a test for this scenario?

**Review by jonathanyeung — CHANGES_REQUESTED** [review]

File: packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts (hunk 2 above)
Comment (Harshita-mindfire) [reply packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts:19]: The string here is the return type when the destination is clipboard(equivalent to JSON publish pod)

File: packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts (hunk 3 above)
Comment (Harshita-mindfire) [reply packages/pods-core/src/v2/pods/export/JSONExportPodV2.ts:67]: When running the export pod, the destination quickpick does not show the clipboard option for multi note export, it directly opens the local file system selector. If the user updates the config with destination as "clipboard", added a check for this to throw an error: 746b330

**Review by jonathanyeung — APPROVED** [review]

**Pull request merged**